Tests/Editor/Unit/Core/EditingContextManagerTests.cs

```diff
@@ -0,0 +1,171 @@
+			context.name = "Some Other Context";
+			context.m_DefaultToolStack = new List<MonoScript>();
+			context.m_DefaultToolStack.Add(MonoScript.FromMonoBehaviour(transformTool));
+			context.m_DefaultToolStack.Add(MonoScript.FromMonoBehaviour(createPrimitiveTool));
+
+			context2 = ScriptableObject.CreateInstance<EditorVRContext>();
+			context2.name = "Yet Another Context";
+			context2.m_DefaultToolStack = context.m_DefaultToolStack;
+
+			settings = ScriptableObject.CreateInstance<EditingContextManagerSettings>();
+			settings.defaultContextName = "Custom Default Context";
+			newSettings = ScriptableObject.CreateInstance<EditingContextManagerSettings>();
+			newSettings.defaultContextName = "New Custom Default Context";
+		}
+
+		[Test]
+		public void Initializes_WithDefaultContext()
+		{
+			Assert.AreEqual(manager.defaultContext, manager.currentContext);
+		}
+
+		[Test]
+		public void Initializes_ISetEditingContextMethods()
+		{
+			Assert.IsNotNull(ISetEditingContextMethods.getAvailableEditingContexts);
+			Assert.IsNotNull(ISetEditingContextMethods.getPreviousEditingContexts);
+			Assert.IsNotNull(ISetEditingContextMethods.setEditingContext);
+			Assert.IsNotNull(ISetEditingContextMethods.restorePreviousEditingContext);
+		}
+
+		[Test]
+		public void SetEditingContext_DoesNothing_IfNull()
+		{
+			var beginningContext = manager.currentContext;
+			manager.SetEditingContext(null);
+			Assert.AreEqual(beginningContext, manager.currentContext);
+		}
+
+		[Test]
+		public void SetEditingContext_SetsCurrentContext_IfNotNull()
+		{
+			manager.SetEditingContext(context);
+			Assert.AreEqual(context, manager.currentContext);
+		}
+
+		[Test]
+		public void RestorePreviousContext_SetsPreviousContextToCurrent()
+		{
+			var beginningContext = manager.currentContext;
+			manager.SetEditingContext(context2);
+			Assert.AreNotEqual(beginningContext, manager.currentContext);
+
+			manager.RestorePreviousContext();
+			Assert.AreEqual(beginningContext, manager.currentContext);
+		}
+
+		[Test]
+		public void LoadProjectSettings_IfAssetFound()
+		{
+			EditingContextManager.SaveProjectSettings(settings);
+			var loaded = EditingContextManager.LoadProjectSettings();
+			Assert.AreEqual(settings.defaultContextName, loaded.defaultContextName);
+		}
+
+		[Test]
+		public void LoadProjectSettings_IfAssetNotFound()
+		{
+			if (File.Exists(EditingContextManager.k_SettingsPath))
+				File.Delete(EditingContextManager.k_SettingsPath);
+
+			var loaded = EditingContextManager.LoadProjectSettings();
+			Assert.IsInstanceOf<EditingContextManagerSettings>(loaded);
+			Assert.IsNull(loaded.defaultContextName);
+		}
+
+		[Test]
+		public void LoadUserSettings_NewerThanProjectSettings()
+		{
+			EditingContextManager.SaveUserSettings(newSettings);
+			var loaded = EditingContextManager.LoadUserSettings();
+			Assert.AreEqual(newSettings.defaultContextName, loaded.defaultContextName);
+		}
+
+		[Test]
+		public void LoadUserSettings_OlderThanProjectSettings()
```

Comment:
What exactly are these two testing?

Reply:
we defer to project settings if they are newer than user settings